src/site/tests/home/00-required.e2e.spec.js

```diff
@@ -0,0 +1,53 @@
+const fs = require('fs');
+const path = require('path');
+
+const E2eHelpers = require('platform/testing/e2e/helpers');
+
+function verifyGoogleAnalytics(browser) {
+  const googleAnalyticsCode = fs.readFileSync(
+    path.join(
+      __dirname,
+      '../../assets/js/google-analytics/',
+      `${process.env.BUILDTYPE}.js`,
+    ),
+  );
+
+  browser.execute(
+    () => document.querySelector('[data-e2e="analytics-script"]').innerHTML,
+    [],
+    ({ value: analyticsScriptHtml }) => {
+      const codeExists = analyticsScriptHtml.includes(googleAnalyticsCode);
+      browser.assert.equal(
+        codeExists,
+        true,
+        'Google Analytics code exists in the DOM.',
+      );
+    },
+  );
+}
```

Comment:
I'm not in love with this code, but Liquid templates are very frail, and if something happened to the `buildtype` metadata, it would be terrible if we lose our analytics even just for a few hours. This code makes sure that the GA snippet is in the DOM.

Reply:
Just for my own curiosity... Even if `buildtype` were missing, it looks like it could still default to the case for production. Is the templating frail to the point it would just completely not include that section altogether?

Reply:
Yeah, this is mostly just paranoia. Metalsmith has a `metadata` function for assigning template globals, but each call to `metadata` wipes the previous call. There have been a couple times where we've seen that erase `buildtype`. I think this E2E test will enforce that the correct `buildtype` is always supplied in our templates.